src/frontend/ts/state/chessStore.ts

```diff
@@ -112,6 +114,58 @@ const useChessStore = create<ChessState>((set, get) => ({
     await get().fetchArchivedMoves(gameID)
     get().setDisplayGame(get().archivedGames.get(gameID))
   },
+  countTallies: async (archivedGames: Map<GameID, ArchivedGameInfo>) => {
```

Comment:
We shouldn't be recomputing every tally for every player each time we receive a new finished game. `tallies` should be a `Map<Ship, Record>`. When we receive a new archived game, we read the result and increment the appropriate field in the `Record` for the opponent `Ship`. Instead of `countTallies`, we have a helper function in `ts/helpers/chess.ts` that will convert a `Record` into a `String`. We call this function on the `.get()` output of `tallies` in `Challenges.tsx`. We don't need to track the numbers for both players because they're implicitly tracked: our wins are the opponents losses, and vice versa.

Reply:
I wrote a more-or-less complete version of what `receiveArchivedGame` should look like for this:
```
const tallies = get().tallies
let opponent = data.white === window.ship
  ? data.black
  : data.white
let record = tallies.has(opponent)
  ? tallies.get(opponent)
  : new Map().set(opponent, { wins: 0, losses: 0, draws: 0 });
let newRecord = null

switch (data.result) {
  case "1-0":
    if (data.white == window.ship) {
      newRecord = {
        ...record,
        wins: record.wins + 1
      }
    } else {
      newRecord = {
        ...record,
        losses: record.losses + 1
      }
    }
    break;
  case "0-1":
    if (data.white == window.ship) {
      newRecord = {
        ...record,
        losses: record.losses + 1
      }
    } else {
      newRecord = {
        ...record,
        wins: record.wins + 1
      }
    }
    break;
  case "½–½":
    newRecord = {
      ...record,
      draws: record.draws + 1
    }
    break;
  default:
    // XX: Need to print message to console and error out
}

set(state => ({
  archivedGames: state.archivedGames.set(data.gameID, data)
  tallies: state.tallies.set(opponent, newRecord)
  }))
```